src/ripple/app/tx/impl/NFTokenModify.cpp

```diff
@@ -0,0 +1,96 @@
+
+#include <ripple/app/tx/impl/NFTokenModify.h>
+#include <ripple/app/tx/impl/details/NFTokenUtils.h>
+#include <ripple/ledger/View.h>
+#include <ripple/protocol/Feature.h>
+#include <ripple/protocol/Rate.h>
+#include <ripple/protocol/TxFlags.h>
+#include <ripple/protocol/st.h>
+
+namespace ripple {
+
+NotTEC
+NFTokenModify::preflight(PreflightContext const& ctx)
+{
+    if (!ctx.rules.enabled(featureDNFT))
+        return temDISABLED;
+
+    if (auto const ret = preflight1(ctx); !isTesSuccess(ret))
+        return ret;
+
+    auto const nftID = ctx.tx[sfNFTokenID];
+    auto const account = ctx.tx[sfAccount];
+
+    if (!account || !nftID)
+       return temMALFORMED; 
+
+    if (ctx.tx.isFieldPresent(sfURI))
+    {
+        auto uri = ctx.tx[sfURI];
+
+        if (uri.length() == 0 || uri.length() > maxTokenURILength)
+            return temMALFORMED;
+    }
+
+    return preflight2(ctx);
+}
+
+TER
+NFTokenModify::preclaim(PreclaimContext const& ctx)
+{
+    auto const account = ctx.tx[sfAccount]; 
+
+    auto const owner = [&ctx]() {
+        if (ctx.tx.isFieldPresent(sfOwner))
+            return ctx.tx.getAccountID(sfOwner);
+
+        return ctx.tx[sfAccount];
+    }();
+
+    if (!nft::findToken(ctx.view, owner, ctx.tx[sfNFTokenID]))
+        return tecNO_ENTRY;
+
+    // Check if the NFT is mutable
+    if (!(nft::getFlags(ctx.tx[sfNFTokenID]) & nft::flagMutable))
+        return tecNO_PERMISSION;
+
+    // Verify permissions for the issuer
+    if (auto const issuer = nft::getIssuer(ctx.tx[sfNFTokenID]);
+        issuer != account)
+    {
+        if (auto const sle = ctx.view.read(keylet::account(issuer)); sle)
+        {
+            if (auto const minter = (*sle)[~sfNFTokenMinter];
+                minter != account)
+                return tecNO_PERMISSION;
+        }
+    }
+
+    return tesSUCCESS;
+}
+
+TER
+NFTokenModify::doApply()
+{
+    auto const nftokenID = ctx_.tx[sfNFTokenID];
+    auto const account = ctx_.tx[sfAccount];
+
+
+    // Find the token and its page
+    auto tokenAndPage = nft::findTokenAndPage(view(), account, nftokenID);
+    if (!tokenAndPage)
+        return tecINTERNAL;
+
+    // Replace the URI if present in the transaction
+    if (ctx_.tx.isFieldPresent(sfURI)) {
+        auto newURI = ctx_.tx[sfURI];
+        tokenAndPage->token[sfURI] = newURI; 
+    }
```

Comment:
Since the URI is a required field in the Standard, this condition is always true if it is followed.

If the URI can be deleted, the URI field should be an optional field.

I think it would be appropriate to change the URI field in Standard to an optional field.
And overwrite the URI if it exists in the transaction and delete the URI if it does not.